src/main/java/org/wildfly/security/sasl/digest/DigestClientFactory.java

```diff
@@ -60,6 +62,9 @@ public DigestClientFactory(final Provider provider) {
     public SaslClient createSaslClient(String[] mechanisms, String authorizationId, String protocol, String serverName,
             Map<String, ?> props, CallbackHandler cbh) throws SaslException {
 
+        if (cbh == null) throw ElytronMessages.log.mechCallbackHandlerMissing().toSaslException();
```

Comment:
We should keep the compatibility with the JDK implementation. Some mechanisms (e.g. `DIGEST-MD5`) reimplement the ones provided by the JDK and the exception handling  should be kept as close as possible to this default.

Example from [com.sun.security.sasl.digest.FactoryImpl class](http://hg.openjdk.java.net/jdk8u/jdk8u/jdk/file/0e4fc29a5ce4/src/share/classes/com/sun/security/sasl/digest/FactoryImpl.java):

```java
if (cbh == null) {
    throw new SaslException(
        "Callback handler with support for RealmChoiceCallback, " +
        "RealmCallback, NameCallback, and PasswordCallback " +
        "required");
}
```